### PR TITLE
added throw-able `ActorRegistry.Register` method

### PR DESCRIPTION
**Pull request by Aaronontheweb**
Wanted to do this to make assertions easier to implement within the `AkkaConfigurationBuilder`.

### CODE REVIEW

File: src/Akka.Hosting/ActorRegistry.cs [hunk 1]
```diff
@@ -30,6 +42,17 @@ public class ActorRegistry : IActorRegistry, IExtension
...
             new ConcurrentDictionary<Type, IActorRef>();
 
+        /// <inheritdoc cref="IActorRegistry.Register{TKey}"/>
+        /// <exception cref="DuplicateActorRegistryException">Thrown when the same value is inserted twice and overwriting is not allowed.</exception>
+        public void Register<TKey>(IActorRef actor, bool overwrite = false)
+        {
+            if (!TryRegister<TKey>(actor, overwrite))
+            {
+                throw new DuplicateActorRegistryException(
+                    $"An actor for type {typeof(TKey)} has already been . Call `Register(IActorRef, bool overwrite=true)` to avoid this error or use a different key.");
```
Comment (Arkatufus) [inline src/Akka.Hosting/ActorRegistry.cs:52]: "has already been ." missing "registered"

File: src/Akka.Hosting/ActorRegistry.cs [hunk 2]
```diff
@@ -160,6 +183,13 @@ public interface IReadOnlyActorRegistry : IEnumerable<KeyValuePair<Type, IActorR
     /// </remarks>
     public interface IActorRegistry: IReadOnlyActorRegistry
     {
+        /// <summary>
+        /// Registers an actor into the registry. Throws an exception upon failure.
+        /// </summary>
+        /// <param name="actor">The bound <see cref="IActorRef"/>, if any. Is set to <see cref="ActorRefs.Nobody"/> if key is not found.</param>
```
Comment (Arkatufus) [inline src/Akka.Hosting/ActorRegistry.cs:189]: So we can register a null value to a key?

**Review by Arkatufus — CHANGES_REQUESTED** [review]
Seems like there are some typo and ambiguity in the XML doc

File: src/Akka.Hosting/ActorRegistry.cs (hunk 1 above)
Comment (Aaronontheweb) [reply src/Akka.Hosting/ActorRegistry.cs:52]: whoops - nice catch

File: src/Akka.Hosting/ActorRegistry.cs (hunk 2 above)
Comment (Aaronontheweb) [reply src/Akka.Hosting/ActorRegistry.cs:189]: You can - nothing preventing that for now.

Comment (Aaronontheweb) [reply src/Akka.Hosting/ActorRegistry.cs:189]: Although this XML-DOC is wrong.

Comment (Arkatufus) [reply src/Akka.Hosting/ActorRegistry.cs:189]: Might want to do a null check and throw a `NullArgumentException`, maybe?

Comment (Aaronontheweb) [reply src/Akka.Hosting/ActorRegistry.cs:189]: Good idea

**Review by Arkatufus — APPROVED** [review]
LGTM

**Pull request merged**